### PR TITLE
Change Story Ad Insertion

### PR DESCRIPTION
**Pull request by calebcordry**
FR was to have the first ad show after 7 pages, and then after every 8 thereafter. 

Tweaked the page counting a bit so that the defined `const`s more accurately reflect the page of insertion.

### CODE REVIEW

File: extensions/amp-story-auto-ads/0.1/amp-story-auto-ads.js [hunk 1]
```diff
@@ -576,13 +586,36 @@ export class AmpStoryAutoAds extends AMP.BaseElement {
...
+    }
+
+    return false;
+  }
 
   /**
    * start the process over
    * @private
    */
   startNextPage_() {
```
Comment (lannka) [inline extensions/amp-story-auto-ads/0.1/amp-story-auto-ads.js]: naming: startNextAdPage_

File: extensions/amp-story-auto-ads/0.1/amp-story-auto-ads.js [hunk 2]
```diff
@@ -30,8 +30,11 @@ import {parseJson} from '../../../src/json';
 import {setStyles} from '../../../src/style';
 import {triggerAnalyticsEvent} from '../../../src/analytics';
 
-/** @const */
-const MIN_INTERVAL = 4;
+/** @const {number} */
+const FIRST_AD_MIN = 7;
```
Comment (lannka) [inline extensions/amp-story-auto-ads/0.1/amp-story-auto-ads.js]: does this mean the 8th page can be an ad?

File: extensions/amp-story-auto-ads/0.1/amp-story-auto-ads.js [hunk 3]
```diff
@@ -30,8 +30,11 @@ import {parseJson} from '../../../src/json';
...
+
+/** @const {number} */
+const MIN_INTERVAL = 8;
```
Comment (lannka) [inline extensions/amp-story-auto-ads/0.1/amp-story-auto-ads.js]: does this mean the 9th page be an ad? or 8th?

File: extensions/amp-story-auto-ads/0.1/amp-story-auto-ads.js [hunk 4]
```diff
@@ -146,8 +149,12 @@ export class AmpStoryAutoAds extends AMP.BaseElement {
     /** @private {?../../amp-story/0.1/navigation-state.NavigationState} */
     this.navigationState_ = null;
 
-    /** @private {number} */
-    this.uniquePagesCount_ = 0;
+    /**
+     * Set to one here because we count on navigation event, and we do have
+     * an event before first page.
```
Comment (lannka) [inline extensions/amp-story-auto-ads/0.1/amp-story-auto-ads.js]: i don't fully understand this. could you rephrase?

File: extensions/amp-story-auto-ads/0.1/amp-story-auto-ads.js [hunk 5]
```diff
@@ -576,13 +586,36 @@ export class AmpStoryAutoAds extends AMP.BaseElement {
...
   }
 
+  /**
+   * Determine if user has seen enough pages to show an ad. We want a certain
+   * number of pages before the first ad, and then a separate interval
+   * thereafter.
+   * @return {boolean}
+   * @private
+   */
+  enoughPagesShown_() {
```
Comment (lannka) [inline extensions/amp-story-auto-ads/0.1/amp-story-auto-ads.js]: enoughContentPagesViewed_()

File: extensions/amp-story-auto-ads/0.1/amp-story-auto-ads.js [hunk 6]
```diff
@@ -182,6 +189,9 @@ export class AmpStoryAutoAds extends AMP.BaseElement {
     /** @private {number|null} */
     this.idOfAdShowing_ = null;
 
+    /** @private {boolean} */
+    this.firstAdShown_ = false;
```
Comment (lannka) [inline extensions/amp-story-auto-ads/0.1/amp-story-auto-ads.js]: let's be consistent about terminology here, i think this mean `viewed`?

File: extensions/amp-story-auto-ads/0.1/amp-story-auto-ads.js [hunk 7]
```diff
@@ -576,13 +586,36 @@ export class AmpStoryAutoAds extends AMP.BaseElement {
...
    * @private
    */
   startNextPage_() {
-    this.uniquePagesCount_ = 0;
+    if (!this.firstAdShown_) {
+      this.firstAdShown_ = true;
+    }
+
+    // Set to -1 because there is still one page to be viewed before ad.
+    this.uniquePagesCount_ = -1;
```
[the remark this answers is not in the document]
Comment (lannka) [reply extensions/amp-story-auto-ads/0.1/amp-story-auto-ads.js]: indeed, still don't quite understand why it was initialized as 1 and reset as -1

File: extensions/amp-story-auto-ads/0.1/amp-story-auto-ads.js (hunk 6 above)
Comment (calebcordry) [reply extensions/amp-story-auto-ads/0.1/amp-story-auto-ads.js]: Done.

File: extensions/amp-story-auto-ads/0.1/amp-story-auto-ads.js (hunk 5 above)
Comment (calebcordry) [reply extensions/amp-story-auto-ads/0.1/amp-story-auto-ads.js]: Done.

File: extensions/amp-story-auto-ads/0.1/amp-story-auto-ads.js (hunk 4 above)
Comment (calebcordry) [reply extensions/amp-story-auto-ads/0.1/amp-story-auto-ads.js]: I tried to explain above. Let me know if you think there is a better way to rephrase.

Comment (calebcordry) [reply extensions/amp-story-auto-ads/0.1/amp-story-auto-ads.js]: Rewrote this. LMK if you think it should still be improved.

**Comment by calebcordry** [comment]
A couple extra things after initial review: 

Moved the logic so that we start next ad loading after view. This eliminates the -1 confusion.

Set pages before first ad, and interval thereafter to be the same. I think we should keep the extra logic in case we want to special case the first ad in the future.

File: extensions/amp-story-auto-ads/0.1/amp-story-auto-ads.js [hunk 8]
```diff
@@ -551,38 +562,66 @@ export class AmpStoryAutoAds extends AMP.BaseElement {
...
   }
 
+  /**
+   * Determine if user has seen enough pages to show an ad. We want a certain
+   * number of pages before the first ad, and then a separate interval
+   * thereafter.
+   * @return {boolean}
+   * @private
+   */
+  enoughContentPagesShown_() {
```
Comment (lannka) [inline extensions/amp-story-auto-ads/0.1/amp-story-auto-ads.js]: enoughContentPagesViewed

Comment (calebcordry) [reply extensions/amp-story-auto-ads/0.1/amp-story-auto-ads.js]: done.
